modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/service/test/JournalArticleServiceTest.java

```diff
@@ -277,6 +280,36 @@ public void testDeleteTemplateReferencedByJournalArticles()
 		}
 	}
 
+	@Test
+	public void testDeleteTemplateSpecifiedByDeletedJournalArticle()
+		throws Exception {
+
+		DDMStructure ddmStructure = DDMStructureTestUtil.addStructure(
+			_group.getGroupId(), JournalArticle.class.getName());
+
+		DDMTemplate ddmTemplate = DDMTemplateTestUtil.addTemplate(
+			_group.getGroupId(), ddmStructure.getStructureId(),
+			PortalUtil.getClassNameId(JournalArticle.class));
+
+		JournalArticle journalArticle =
+			JournalTestUtil.addArticleWithXMLContent(
+				_group.getGroupId(), "<title>Test Article</title>",
+				ddmStructure.getStructureKey(), ddmTemplate.getTemplateKey());
+
+		String compositeClassName = ResourceActionsUtil.getCompositeModelName(
+			JournalArticle.class.getName(), DDMTemplate.class.getName());
+
+		_ddmTemplateLinkLocalService.addTemplateLink(
+			_classNameLocalService.getClassNameId(compositeClassName),
+			journalArticle.getId(), ddmTemplate.getTemplateId());
+
+		_journalArticleLocalService.deleteArticle(
+			_group.getGroupId(), journalArticle.getArticleId(),
+			new ServiceContext());
+
+		_ddmTemplateLocalService.deleteTemplate(ddmTemplate.getTemplateId());
```

Comment:
@jonathanmccann, just some quick questions about this test: we only check that the methods don't throw errors? We don't assert that the templates are no longer present?

Reply:
@jkappler Yes, that was the intention. If it's better to add an assert to make it clearer I can add that as well.

Reply:
@jonathanmccann, I think it would be better to assert that the template links were deleted. Mostly if someone else in the future arrives to this test and decides to change it 😅

Reply:
Thanks for the feedback. I've pushed up the changes but let me know if there is anything else that needs to be reworked.